### PR TITLE
LPS-154515 Add feature flag for Epic scope

### PR DESCRIPTION
**Pull request by ces-luannguyen**
Hi @tototrinh @locpham97 @ces-quanhuynh 
Please help me review this PR.

### Motivation
LPS-184810 Add/Remove Feature or Release flag of LPS-154515
### Solution
Modify portal.properties in portal-impl module to show toggle box for enabling/disabling Epic feature in Instance Settings > Feature Flags. This configuration is read to enable/disable code used in epic.

Thanks,
Luan Nguyen

### CODE REVIEW

File: portal-impl/src/portal.properties [hunk 1]
```diff
@@ -6778,6 +6783,22 @@
...
 
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASES__MINUS__NUMBER1__NUMBER5__NUMBER4__NUMBER5__NUMBER1__NUMBER5__PERIOD_DESCRIPTION
+    #
+    feature.flag.LPS-154515.description=Documents manager will be able to set permissions from folders to folders by inheritance.
+
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASES__MINUS__NUMBER1__NUMBER5__NUMBER4__NUMBER5__NUMBER1__NUMBER5__PERIOD_TITLE
+    #
+    feature.flag.LPS-154515.title=Permission propagation
```
Comment (ces-quanhuynh) [inline portal-impl/src/portal.properties:6794]: Should capital the Propagation word. Permission Propagation

File: portal-impl/src/portal.properties [hunk 2]
```diff
@@ -6778,6 +6783,22 @@
     #
     feature.flag.LPS-134060.type=beta
 
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASES__MINUS__NUMBER1__NUMBER5__NUMBER4__NUMBER5__NUMBER1__NUMBER5__PERIOD_DESCRIPTION
+    #
+    feature.flag.LPS-154515.description=Documents manager will be able to set permissions from folders to folders by inheritance.
```
Comment (ces-quanhuynh) [inline portal-impl/src/portal.properties:6789]: I think the message should be this: 
Enable documents manager to propagate folders' permissions to their children folders.

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 3]
```diff
@@ -142,12 +143,15 @@ public DLFolder addFolder(
 
 		// Permission Propagation
 
-		_permissionPropagationEntryLocalService.addPermissionPropagationEntry(
-			user.getCompanyId(), groupId, DLFolder.class.getName(),
-			dlFolder.getFolderId(),
-			ParamUtil.getBoolean(
-				serviceContext, "permissionPropagationEnabled"));
+		if (FeatureFlagManagerUtil.isEnabled("LPS-154515")) {
```
Comment (ces-quanhuynh) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java:147]: delete empty line here

File: modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/portlet/PortletConfigurationPortlet.java [hunk 4]
```diff
@@ -597,17 +598,19 @@ public void updateRolePermissions(
 			}
 		}
 
-		PortletConfigurationPermissionPropagation
-			portletConfigurationPermissionPropagation =
+		if (FeatureFlagManagerUtil.isEnabled("LPS-154515")) {
+
```
Comment (ces-quanhuynh) [inline modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/portlet/PortletConfigurationPortlet.java:602]: delete empty line here.

File: modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp [hunk 5]
```diff
@@ -38,7 +38,7 @@ PortletConfigurationPermissionPropagation portletConfigurationPermissionPropagat
 			selectable="<%= false %>"
 		/>
 
-		<c:if test="<%= (portletConfigurationPermissionPropagation != null) && portletConfigurationPermissionPropagation.isAvailable(renderRequest) %>">
+		<c:if test='<%= FeatureFlagManagerUtil.isEnabled("LPS-154515") && (portletConfigurationPermissionPropagation != null) && portletConfigurationPermissionPropagation.isAvailable(renderRequest) %>'>
```
Comment (tototrinh) [inline modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp:41]: How about returning null [here](https://github.com/tototrinh/liferay-portal/blob/LPS-154515/modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/display/context/PortletConfigurationPermissionsDisplayContext.java#L311-L316) if the feature is not enabled?

**Review by ces-totrinh — COMMENTED** [review]
Please run SF. Thanks

**Comment by ces-totrinh** [comment]
How about creating a feature flag for the whole epic instead of each user story and with the type of development until all the rest of the user stories are completed?
@ces-luannguyen @ces-quanhuynh @locpham97

**Comment by ces-quanhuynh** [comment]
Hi @tototrinh 
We updated the code. 
Please review again!
cc @ces-luannguyen

**Comment by ces-totrinh** [comment]
@ces-quanhuynh  Please commit the code under your name and resend. Thanks.

**Pull request closed without merging**